python/global_search_widget/global_search_widget.py

```diff
@@ -8,67 +8,41 @@
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
-import os
 import sgtk
+from sgtk.platform.qt import QtCore, QtGui
 
-shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
-shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
+global_search_copmleter = sgtk.platform.current_bundle().import_module(
+    "global_search_completer")
 
-from sgtk.platform.qt import QtCore, QtGui
-from sgtk import TankError
- 
 class GlobalSearchWidget(QtGui.QLineEdit):
     """
     A QT Widget deriving from :class:`~PySide.QtGui.QLineEdit` that creates
     a global search input box with auto completion.
     
-    :signal entity_selected(str, int): Fires when someone selects an entity inside
+    :signal: ``entity_selected(str, int)`` - Fires when someone selects an entity inside
             the search results. The returned parameters are entity type and entity id.
     """
-    
-    # custom roles for the model that tracks the auto completion results
-    SG_DATA_ROLE = QtCore.Qt.UserRole + 1
-    MODE_ROLE = QtCore.Qt.UserRole + 2
-    
-    # different items in the auto complete list can have 
-    # a different meaning, so track those here too
-    (MODE_LOADING, MODE_NOT_FOUND, MODE_RESULT) = range(3)
-    
+
     # emitted when shotgun has been updated
     entity_selected = QtCore.Signal(str, int)
-    
+
     def __init__(self, parent):
         """
         :param parent: Qt parent object
         :type parent: :class:`~PySide.QtGui.QWidget`        
         """
  
         # first, call the base class and let it do its thing.
-        QtGui.QLineEdit.__init__(self, parent)
+        super(GlobalSearchWidget, self).__init__(parent)
 
-        # set up some handy references
-        self._bundle = sgtk.platform.current_bundle()
-        
-        self.__sg_data_retriever = None
-        
-        self._processing_id = None
-        self._thumb_map = {}
-        self._default_icon = QtGui.QPixmap(":/tk_framework_qtwidgets.global_search_widget/rect_512x400.png")
-                
         # configure our popup completer
-        self._completer = QtGui.QCompleter(self)
-        self._completer.setMaxVisibleItems(10)
-        self._completer.setCaseSensitivity(QtCore.Qt.CaseInsensitive)
-        self._completer.setCompletionMode(QtGui.QCompleter.UnfilteredPopupCompletion)
-        self.setCompleter(self._completer)
+        self.setCompleter(global_search_copmleter.GlobalSearchCompleter(self))
```

Comment:
wow! spelled it wrong on the import too. double whammy. well, it was working!!! fixed.